src/vqcsim/parametric_gate.cpp

```diff
@@ -0,0 +1,469 @@
+#include "parametric_gate.hpp"
+
+QuantumGate_SingleParameter::QuantumGate_SingleParameter(double angle)
+    : _angle(angle) {
+    _parameter_id = OLD_STYLE_PARAMETER;
+    _parameter_coef = 1.;
+    _gate_property |= FLAG_PARAMETRIC;
+}
+QuantumGate_SingleParameter::QuantumGate_SingleParameter(
+    const ParameterId& parameter_id, double parameter_coef)
+    : _parameter_id(parameter_id), _parameter_coef(parameter_coef) {
+    _gate_property |= FLAG_PARAMETRIC;
+}
+
+bool QuantumGate_SingleParameter::is_old_style() const {
+    return _parameter_id == OLD_STYLE_PARAMETER;
+}
+bool QuantumGate_SingleParameter::is_new_style() const {
+    return _parameter_id != OLD_STYLE_PARAMETER;
+}
+
+ParameterId QuantumGate_SingleParameter::get_parameter_id() const {
+    return _parameter_id;
+}
+void QuantumGate_SingleParameter::set_parameter_id(
+    const ParameterId& parameter_id) {
+    if (!is_new_style()) {
+        throw NotImplementedException(
+            "Error: "
+            "QuantumGate_SingleParameter::set_parameter_id(const "
+            "ParameterId&): "
+            "this is a new-style function. You cannot use this function "
+            "with an old-style parametric_gate whose parameter_id is -1");
+    }
+    _parameter_id = parameter_id;
+}
+double QuantumGate_SingleParameter::get_parameter_coef() const {
+    return _parameter_coef;
+}
+double QuantumGate_SingleParameter::get_parameter_value() const {
+    if (!this->is_old_style()) {
+        throw NotImplementedException(
+            "Error: "
+            "QuantumGate_SingleParameter::set_parameter_value(double): "
+            "this is an old-style function. You cannot use this function "
+            "with a new-style parametric_gate whose parameter_id is not -2");
```

Comment:
`OLD_STYLE_PARAMETER`が`-1`なので、`parameter_id`としては`-1`以外を使ってほしいって感じですよね

```suggestion
            "with a new-style parametric_gate whose parameter_id is not -1");
```